strawberryfields/backends/tfbackend/tests/test_fock_probs_gaussian_states.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Unit tests for probabilities of Fock outcomes of Gaussian states"""
```

Comment:
I'm not sure this test is really necessary. We already have separate tests that extensively check:

1. The resulting Fock state of various state preparations

2. The `state.fock_prob` method

This test is essentially repeating those tests, but our filename conventions make it hard to tell without delving into the files themselves...

Reply:
I believe this is a test file that @nquesada wrote, with the intention of verifying these same things for the Gaussian backend. Do those other tests cover Gaussian backend?

Reply:
Yup, that is correct. They are meant to check we do get the right Fock state probs when using a Gaussian state and then effectively calculating a hafnian or using the explicit expression for single mode states. It is odd it is now in the tfbackend.

Reply:
The good news is that the tests pass with TFBackend even if you didn't intend it that way :)

Reply:
ah, so it's just a case of different people writing different tests for the same thing at different times. Something we'll want to merge when/if we refactor the tests.